nncf/torch/sparsity/movement/algo.py

```diff
@@ -0,0 +1,480 @@
+        #     return CompressionStage.FULLY_COMPRESSED
+        if self.scheduler.current_epoch < self.scheduler.warmup_start_epoch:
+            return CompressionStage.UNCOMPRESSED
+        if self.scheduler.current_sparsity_level >= self.scheduler.warmup_end_epoch:
+            return CompressionStage.FULLY_COMPRESSED
+        return CompressionStage.PARTIALLY_COMPRESSED
+
+    def freeze(self):
+        self._loss.disable()
+
+    def distributed(self):
+        if not dist.is_initialized():
+            raise KeyError('Could not set distributed mode for the compression algorithm '
+                           'because the default process group has not been initialized.')
+
+        if next(self._model.parameters()).is_cuda:
+            state = torch.cuda.get_rng_state()
+            if dist.get_backend() == dist.Backend.NCCL:
+                state = state.cuda()
+            torch.distributed.broadcast(state, src=0)
+            torch.cuda.set_rng_state(state.cpu())
+        else:
+            state = torch.get_rng_state()
+            torch.distributed.broadcast(state, src=0)
+            torch.set_rng_state(state)
+
+        self._distributed = True
+
+    def _check_distributed_masks(self):
+        if not self._distributed or get_world_size() == 1:
+            return 1
+
+        nvalues = 0
+        ncor_values = 0
+        eps = 1e-4
+        for minfo in self.sparsified_module_info:
+            mask = minfo.operand.mask
+
+            mask_list = [torch.empty_like(mask) for _ in range(get_world_size())]
+            # nccl does not support gather, send, recv operations
+            dist.all_gather(mask_list, mask)
+
+            for i in range(1, len(mask_list)):
+                rel_error = (mask_list[0] - mask_list[i]) / mask_list[0]
+                ncor_values = ncor_values + (rel_error.abs() < eps).sum(dtype=mask.dtype)
+                nvalues = nvalues + mask_list[i].numel()
+
+        return ncor_values / nvalues
+
+    def statistics(self, quickly_collected_only=False) -> NNCFStatistics:
+        collector = PTSparseModelStatisticsCollector(self.model, self.sparsified_module_info)
+        model_statistics = collector.collect()
+
+        stats = MovementSparsityStatistics(model_statistics,
+                                           self.scheduler.current_importance_threshold,
+                                           self.scheduler.current_importance_lambda)
+
+        nncf_stats = NNCFStatistics()
+        nncf_stats.register('movement_sparsity', stats)
+        return nncf_stats
+
+    def reset_independent_structured_mask(self):
+        self._structured_mask_handler.update_independent_structured_mask()
+
+    def resolve_structured_mask(self):
+        self._structured_mask_handler.resolve_dependent_structured_mask()
+
+    def populate_structured_mask(self):
+        self._structured_mask_handler.populate_dependent_structured_mask_to_operand()
+
+    def report_structured_sparsity(self, dirname):
```

Comment:
put this into DEBUG mode

Reply:
Yujie moved `report_structured_sparsity` into structured mask handler and tests have been developed.